assembl/static2/js/app/components/administration/voteSession/tokensForm.jsx

```diff
@@ -0,0 +1,126 @@
+// @flow
+import React from 'react';
+import { connect } from 'react-redux';
+import { I18n } from 'react-redux-i18n';
+import { Checkbox, DropdownButton, MenuItem } from 'react-bootstrap';
+import FormControlWithLabel from '../../common/formControlWithLabel';
+import { getEntryValueForLocale } from '../../../utils/i18n';
+import Helper from '../../common/helper';
+import TokenTypeForm from './tokenTypeForm';
+import {
+  updateTokenVoteInstructions,
+  createTokenVoteType,
+  deleteTokenVoteType,
+  updateTokenVoteExclusive
+} from '../../../actions/adminActions/voteSession';
+
+type TokensFormProps = {
+  instructions: string,
+  exclusive: boolean,
+  tokenTypeNumber: number,
+  tokenTypes: Object,
+  editLocale: string,
+  handleInstructionsChange: Function,
+  handleTokenVoteTypeNumberChange: Function,
+  handleExclusiveCheckboxChange: Function
+};
+
+const DumbTokensForm = ({
+  instructions,
+  exclusive,
+  tokenTypeNumber,
+  tokenTypes,
+  editLocale,
+  handleInstructionsChange,
+  handleTokenVoteTypeNumberChange,
+  handleExclusiveCheckboxChange
+}: TokensFormProps) => (
+  <div className="token-vote-form">
+    <form>
+      <div className="flex">
+        <Checkbox
+          checked={exclusive}
+          onChange={() => {
+            handleExclusiveCheckboxChange(exclusive);
```

Comment:
you could just use `handleExclusiveCheckboxChange` and modify `mapDispatchToProps` since you have `exclusive` in `ownProps`

Reply:
I'm not sure to understand. Should I use `mergeProps`?

Reply:
no, it's just that you can use `onChange={handleExclusiveCheckboxChange}` and use `exclusive` from `ownProps` (the second param of `mapDispatchToProps`) in `mapDispatchToProps` (but it's really not a big deal)